src/vs/platform/clipboard/browser/clipboardService.ts

```diff
@@ -28,31 +34,64 @@ export class BrowserClipboardService implements IClipboardService {
 		try {
 			return await navigator.clipboard.writeText(text);
 		} catch (error) {
-			console.error(error);
-		}
-
-		// Fallback to textarea and execCommand solution
-
-		const activeElement = document.activeElement;
-
-		const textArea: HTMLTextAreaElement = document.body.appendChild($('textarea', { 'aria-hidden': true }));
-		textArea.style.height = '1px';
-		textArea.style.width = '1px';
-		textArea.style.position = 'absolute';
+			if (!isSafari) {
+				console.error(error);
+			} else {
+				const showResult = await this.dialogService.show(
```

Comment:
My only worry is that we might have tests that copy something to clipboard (I know notebook tests do because I always end up with a "paragraph 5" in my clipboard after running integration tests) and if that fails a dialog might show modal, not sure that would make tests stuck tbh.

Reply:
Yeah it's a good point. I'll try to run a build like 10 more times to check flakiness